contracts/extensions/ReputationBootstrapper.sol

```diff
@@ -0,0 +1,148 @@
+  uint256 constant DECAY_DENOMINATOR =     1000000000000000;
+
+  // Events
+
+  event GrantSet(bytes32 hashedSecret, uint256 reputationAmount);
+  event GrantClaimed(address recipient, uint256 reputationAmount, uint256 tokenAmount);
+
+  // Data structures
+
+  struct Grant {
+    uint256 amount;
+    uint256 timestamp;
+  }
+
+  // Storage
+
+  address public token;
+  bool public giveTokens;
+
+  mapping (bytes32 => Grant) public grants;
+
+  // Modifiers
+
+  modifier onlyRoot() {
+    require(colony.hasUserRole(msgSender(), 1, ColonyDataTypes.ColonyRole.Root), "reputation-bootsrapper-caller-not-root");
+    _;
+  }
+
+  // Overrides
+
+  /// @notice Returns the identifier of the extension
+  function identifier() public override pure returns (bytes32) {
+    return keccak256("ReputationBootstrapper");
+  }
+
+  /// @notice Returns the version of the extension
+  function version() public override pure returns (uint256) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    require(address(colony) == address(0x0), "extension-already-installed");
+
+    colony = IColony(_colony);
+    token = colony.getToken();
+  }
+
+  /// @notice Called when upgrading the extension
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension
+  function deprecate(bool _deprecated) public override auth {
+    deprecated = _deprecated;
+  }
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
+    uint256 balance = ERC20(token).balanceOf(address(this));
+    require(ERC20(token).transfer(address(colony), balance), "reputation-bootstrapper-transfer-failed");
+
+    selfdestruct(address(uint160(address(colony))));
+  }
+
+  // Public
+
+  function setGiveTokens(bool _giveTokens) public onlyRoot {
+    giveTokens = _giveTokens;
+  }
+
+  function setGrants(bytes32[] memory _hashedSecrets, uint256[] memory _amounts) public onlyRoot notDeprecated {
+    require(_hashedSecrets.length == _amounts.length, "reputation-bootsrapper-invalid-arguments");
+
+    for (uint256 i; i < _hashedSecrets.length; i++) {
+      require(_amounts[i] <= INT128_MAX, "reputation-bootstrapper-invalid-amount");
+      grants[_hashedSecrets[i]] = Grant(_amounts[i], block.timestamp);
+
+      emit GrantSet(_hashedSecrets[i], _amounts[i]);
+    }
+  }
+
+  function claimGrant(uint256 _secret) public {
+    bytes32 hashedSecret = keccak256(abi.encodePacked(_secret));
+    uint256 grantAmount = grants[hashedSecret].amount;
+    uint256 tokenAmount = grants[hashedSecret].amount;
+    uint256 grantTimestamp = grants[hashedSecret].timestamp;
+
+    require(grantAmount > 0, "reputation-bootstrapper-nothing-to-claim");
+
+    delete grants[hashedSecret];
+
+    // First, decay by any full days
```

Comment:
Ok, I see what you're getting at, if we cut the time in half and double the decay factor we'll be able to apply only a handful of calculations rather than all of them. So for a 50 hour decay we'd end up applying essentially `(k**2)(k**16)(k**32)` which is equivalent to `(k**50)`.